cyral/resource_cyral_sidecar_listener.go

```diff
@@ -402,5 +418,22 @@ func getSidecarListenerSchema() map[string]*schema.Schema {
 				},
 			},
 		},
+		OverrideRepoClientTlsSettingsKey: {
+			Description: "Override TLS settings defined in the repo. " +
+				"Default value generated by API if not provided. ",
+			Type:     schema.TypeBool,
+			Optional: true,
+			Computed: true,
+		},
+		TlsModeKey: {
```

Comment:
@gengdahlCyral could you please also add validation for the possible values that are valid for this field? I understand that the API already does this validation and that it is easier to maintain if we don't duplicate the validation here in the terraform provider, but since the validation messages returned by the API are often not clear enough, adding a validation that is computed during the terraform plan instead of during the execution, and that also provides a proper message that explicitly mentions the error and which values are allowed for this field would improve the terraform provider UX overall, which is something we are aiming for in a recent conversation that I had with @wcmjunior. The validation could be something like:
```Go
ValidateFunc: validation.StringInSlice(tlsModes(), false),
```